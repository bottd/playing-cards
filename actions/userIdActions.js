export const actionTypes = {
  resetUserID: 'RESET_USER_ID',
  setUserID: 'SET_USER_ID',
};

export const resetUserID = () => ({
  type: actionTypes.resetUserID,
});

export const setUserID = id => ({
  type: actionTypes.setUserID,
  id,
});
