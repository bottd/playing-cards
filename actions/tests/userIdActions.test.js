import { actionTypes, resetUserID, setUserID } from '../userIdActions';

describe('userIdActions', () => {
  describe('Action types', () => {
    it('Should match snapshot', () => {
      expect(actionTypes).toMatchSnapshot();
    });
  });

  describe('resetUserID', () => {
    it('Should return an action with correct type', () => {
      const expected = { type: 'RESET_USER_ID' };
      const result = resetUserID();
      expect(result).toEqual(expected);
    });
  });

  describe('setUserID', () => {
    it('Should return an action with correct type', () => {
      const expected = 'SET_USER_ID';
      const result = setUserID(1);
      expect(result.type).toBe(expected);
    });
    it('Should have the correct id inside the action', () => {
      const expected = 1;
      const result = setUserID(expected);
      expect(result.id).toBe(expected);
    });
    it('Should match snapshot', () => {
      const result = setUserID(1);
      expect(result).toMatchSnapshot();
    });
  });
});
