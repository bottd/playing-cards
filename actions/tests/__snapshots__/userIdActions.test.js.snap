// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userIdActions Action types Should match snapshot 1`] = `
Object {
  "resetUserID": "RESET_USER_ID",
  "setUserID": "SET_USER_ID",
}
`;

exports[`userIdActions setUserID Should match snapshot 1`] = `
Object {
  "id": 1,
  "type": "SET_USER_ID",
}
`;
